### ISSUE TITLE
Inconsistent Handling of Failed EVM Transactions in JSON-RPC Relay

### ISSUE THREAD

**Issue by mgarbs** [hashgraph/hedera-json-rpc-relay#3115]
Title: 

## Description:

I'm encountering an issue where the same EVM transaction appears in consecutive blocks on the Hedera testnet, and the JSON-RPC relay handles these duplicates inconsistently. Specifically, a transaction that fails due to WRONG_NONCE is included in the next block but is misrepresented as successful in both the JSON-RPC responses and the block explorer.

## Steps to Reproduce:

Observe the Duplicate Transactions:

Visit the following block in HashScan: [Block 9337619](https://hashscan.io/testnet/block/9337619)
Notice that the EVM transaction appears twice:
[Transaction Success](https://hashscan.io/testnet/transaction/1726642023.105113000)
[Transaction Failure](https://hashscan.io/testnet/transaction/1726642023.211917000)

Query the Transaction Receipt by Hash:
```bash
curl -s https://testnet.hashio.io/api \
  --header "Content-Type: application/json" \
  --data '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0xb3390118380397cb4e483d01aa0f2f95fef2d3b2ce681fac5c60b5a7ee34492c"],"id":1}' | jq
```

The receipt indicates success, even for the failed transaction.
Retrieve EVM Transactions in Block 9337619:
```bash
curl -s https://testnet.hashio.io/api \
  --header 'Content-Type: application/json' \
  --data '{"jsonrpc":"2.0","id":1,"method":"eth_getBlockByNumber","params":["0x8e7b13", true]}' | jq
```

Only one instance of the transaction is returned, despite there being two occurrences.
Retrieve EVM Transactions in Block 9337620 (Next Block):

```bash
curl -s https://testnet.hashio.io/api \
  --header 'Content-Type: application/json' \
  --data '{"jsonrpc":"2.0","id":1,"method":"eth_getBlockByNumber","params":["0x8e7b14", true]}' | jq
```

The failed duplicate transaction is included in this block as well.

## Expected Behavior:
Failed EVM transactions due to WRONG_NONCE should not be included in subsequent blocks or should be accurately represented as failed in the JSON-RPC responses.
The JSON-RPC relay should not misrepresent the status of failed transactions.
Fees should not be charged for transactions that fail due to WRONG_NONCE.

## Actual Behavior:
The same EVM transaction appears in two consecutive blocks.
The failed transaction is included and misrepresented as successful in the JSON-RPC responses and on the block explorer.
Events emitted during the failed transaction are displayed, causing confusion.
It's unclear whether fees are being charged for the failed transaction.

## Impact:
This issue affects transaction receipt and event validation, which are crucial for cross-chain message validation.
Applications relying on accurate transaction data may experience inconsistencies or erroneous behavior.

## Additional Information:
Transaction Hash: 0xb3390118380397cb4e483d01aa0f2f95fef2d3b2ce681fac5c60b5a7ee34492c
Blocks Affected: 9337619 and 9337620 (Testnet)
RPC used: https://testnet.hashio.io/api

## Relevant Links:
[HashScan Block 9337619](https://hashscan.io/testnet/block/9337619)
[Successful Transaction Details](https://hashscan.io/testnet/transaction/1726642023.105113000)
[Failed Transaction Details](https://hashscan.io/testnet/transaction/1726642023.211917000)

## Questions:
Can failed EVM transactions be excluded from JSON-RPC responses to prevent confusion?:

Please investigate this issue to ensure that:

Failed EVM transactions are accurately represented in JSON-RPC responses.
Duplicate transactions are handled consistently across blocks.

**Comment by victor-yanev** [comment]
CC @steven-sheehy @xin-hedera

**Comment by steven-sheehy** [comment]
I don't see any mirror node issue. You might be confusing the Hedera transaction hash and Ethereum transaction hash. If the user submitted the same Ethereum transaction 3 times then they should each have unique Hedera hashes but the same Ethereum transaction hash. The first will succeed and the following 2 will fail due to wrong nonce. The third one happened to occur in a separate block so it would have a different block number.

Mirror node returns all successful or attempted contract executing transactions in the contract results. This is expected and we can't change this behavior. Relay seems to collapse duplicates but doesn't take into account the status. It should change to prioritize keeping successful hashes if present and discarding unsuccessful transactions.

**Comment by victor-yanev** [comment]
@steven-sheehy Fair point, I didn’t see there was a third duplicate. We will fix the issue on our side by collapsing the failed transactions. Thanks for the fast response.
